statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,569 @@
+                if (self.ncs[s] == 1 or self.ncs[s] == 2):
+                    continue
+                # array of clusters to resample from
+                ii = np.flatnonzero(self.sfclust == s)
+                # resample them
+                ii_resample = np.random.choice(ii, size=(self.ncs[s]-1))
+                # accumulate number of times cluster i was resampled
+                clust_count += np.bincount(ii_resample,
+                                   minlength=max(self.sclust)+1)
+        # avg number of times cluster i was resampled
+        clust_count *= (bsn / replicates)
+        # augment weights
+        w *= clust_count[self.sclust]
+        return w
+
+
+class SurveyStat(SurveyDesign):
+    """
+    Estimation and inference for summary statistics in complex surveys.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+
+    Attributes
+    ----------
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vc : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+
+    def __init__(self, design, mse):
+        self.design = design
+        self.mse = mse
+
+    def bootstrap(self, replicates, index=None, bsn=None):
+        """
+        Calculates bootstrap standard errors
+
+        Parameters
+        ----------
+        stat : object
+            Object of class SurveyMean, SurveyTotal, SurveyPercentile, etc
+        replicates : integer
+            The number of replicates that the user wishes to specify
+
+        Returns
+        -------
+        est : ndarray
+            The point estimates of the statistic, calculated on the columns
+            of data.
+        vc : ndarray
+            The variance-covariance of the estimates.
+        """
+        if index is not None:
+            est = self._stat(self.design.weights, index)
+        else:
+            est = self._stat(self.design.weights)
+
+        jdata = []
+        if (self.design.se_method == "boot"):
+            for i in range(replicates):
+                w = self.design._bootstrap_weights()
+                if index is None:
+                    jdata.append(self._stat(weights=w))
+                else:
+                    jdata.append(self._stat(w, index))
+        else:
+            w = self.design._mean_bootstrap_weight(replicates, bsn)
+            if index is None:
+                jdata.append(self._stat(weights=w))
+            else:
+                jdata.append(self._stat(w, index))
+
+        jdata = np.asarray(jdata)
+        if self.mse:
+            print("mse specified")
+            var = np.dot((jdata-est).T, (jdata-est)) / replicates
+            var = np.sqrt(np.diag(var))
+        else:
+            boot_mean = jdata.mean(0)
+            var = ((jdata - boot_mean)**2).sum(0) / replicates
+
+        return est, var
+
+    def jack(self, index=None):
```

Comment:
Since this is now called automatically, not by user, it should be privatized to `_jack`, same with bootstrap.